### ISSUE TITLE
Force overriding to disable JTA and datasources doesn't work anymore in 0.2.0-1

### ISSUE THREAD

**Issue by pzakpzak** [divinespear/jpa-schema-gradle-plugin#24]
I use Wildfly 10.0.0.Final with Hibernate 5.0.7.Final. 
Generating schema using plugin works well in 0.1.12-1 but has some problems with JTA Data Source in 0.2.0-1.

Probably fix in #13 doesn't work anymore.

**Here is my persistence.xml:**

```
<persistence xmlns="http://xmlns.jcp.org/xml/ns/persistence"
             xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
             xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/persistence
             http://xmlns.jcp.org/xml/ns/persistence/persistence_2_1.xsd"
             version="2.1">

    <persistence-unit name="PostgreSqlDS" transaction-type="JTA">
        <provider>org.hibernate.jpa.HibernatePersistenceProvider</provider>
        <jta-data-source>java:jboss/datasources/PostgreSqlDS</jta-data-source>

        <properties>
            <property name="javax.persistence.schema-generation.database.action" value="none"/>
            <property name="javax.persistence.schema-generation.create-source" value="script"/>
            <property name="javax.persistence.schema-generation.drop-source" value="script"/>
            <property name="javax.persistence.schema-generation.create-script-source" value="META-INF/sql/create.sql"/>
            <property name="javax.persistence.schema-generation.drop-script-source" value="META-INF/sql/drop.sql"/>

            <!-- HIBERNATE SETTINGS -->
            <property name="hibernate.hbm2ddl.import_files_sql_extractor" value="org.hibernate.tool.hbm2ddl.MultipleLinesSqlCommandExtractor"/>
            <property name="hibernate.ejb.naming_strategy" value="org.hibernate.cfg.DefaultComponentSafeNamingStrategy"/>
            <property name="hibernate.enable_lazy_load_no_trans" value="true"/>
            <property name="hibernate.dialect" value="com.somepackage.CustomPostgreSql9Dialect"/>
            <property name="hibernate.jdbc.batch_size" value="100" />
            <property name="hibernate.jdbc.order_updates" value="true" />
        </properties>
    </persistence-unit>

</persistence>
```

**generateSchema task (0.1.12-1):**

```
generateSchema {
    skip = false
    format = true
    scanTestClasses = false
    persistenceUnitName = "PostgreSqlDS"
    databaseAction = "none"
    scriptAction = "drop-and-create"
    outputDirectory = new File(projectDir.absolutePath + '/build/classes/main/META-INF', 'sql')
    createOutputFileName = "create.sql"
    dropOutputFileName = "drop.sql"
    createSourceMode = "metadata"
    dropSourceMode = "metadata"
    databaseProductName = "PostgreSQL"
    databaseMajorVersion = 9
    databaseMinorVersion = 5
    namingStrategy = "org.hibernate.cfg.ImprovedNamingStrategy"
    dialect = "com.somepackage.CustomPostgreSql9Dialect"
}
```

**generateSchema task (0.2.0-1):**

```
generateSchema {
    skip = false
    format = true
    scanTestClasses = false
    persistenceUnitName = "PostgreSqlDS"
    databaseAction = "none"
    scriptAction = "drop-and-create"
    outputDirectory = new File(projectDir.absolutePath + '/build/classes/main/META-INF', 'sql')
    createOutputFileName = "create.sql"
    dropOutputFileName = "drop.sql"
    createSourceMode = "metadata"
    dropSourceMode = "metadata"
    databaseProductName = "PostgreSQL"
    databaseMajorVersion = 9
    databaseMinorVersion = 5
    properties = [
            "hibernate.dialect": "com.somepackage.CustomPostgreSql9Dialect",
            "hibernate.namingStrategy": "org.hibernate.cfg.ImprovedNamingStrategy"
    ]
}
```

```
* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':entities:generateSchema'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:203)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:185)
        at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:66)
        at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:50)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
        at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
Caused by: javax.persistence.PersistenceException: [PersistenceUnit: PostgreSqlDS] Unable to build Hibernate SessionFactory
        at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.persistenceException(EntityManagerFactoryBuilderImpl.java:954)
        at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.generateSchema(EntityManagerFactoryBuilderImpl.java:864)
        at org.hibernate.jpa.HibernatePersistenceProvider.generateSchema(HibernatePersistenceProvider.java:155)
        at io.github.divinespear.gradle.plugin.JpaSchemaGenerateTask.defaultGenerate(JpaSchemaGenerateTask.groovy:369)
        at io.github.divinespear.gradle.plugin.JpaSchemaGenerateTask$_generate_closure12.doCall(JpaSchemaGenerateTask.groovy:354)
        at io.github.divinespear.gradle.plugin.JpaSchemaGenerateTask.generate(JpaSchemaGenerateTask.groovy:331)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
        at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.doExecute(AnnotationProcessingTaskFactory.java:227)
        at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:220)
        at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:209)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:585)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:568)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
        ... 14 more
Caused by: org.hibernate.service.spi.ServiceException: Unable to create requested service [org.hibernate.engine.jdbc.env.spi.JdbcEnvironment]
        at org.hibernate.service.internal.AbstractServiceRegistryImpl.createService(AbstractServiceRegistryImpl.java:244)
        at org.hibernate.service.internal.AbstractServiceRegistryImpl.initializeService(AbstractServiceRegistryImpl.java:208)
        at org.hibernate.service.internal.AbstractServiceRegistryImpl.getService(AbstractServiceRegistryImpl.java:189)
        at org.hibernate.engine.jdbc.internal.JdbcServicesImpl.configure(JdbcServicesImpl.java:51)
        at org.hibernate.boot.registry.internal.StandardServiceRegistryImpl.configureService(StandardServiceRegistryImpl.java:94)
        at org.hibernate.service.internal.AbstractServiceRegistryImpl.initializeService(AbstractServiceRegistryImpl.java:217)
        at org.hibernate.service.internal.AbstractServiceRegistryImpl.getService(AbstractServiceRegistryImpl.java:189)
        at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.handleTypes(MetadataBuildingProcess.java:352)
        at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.complete(MetadataBuildingProcess.java:111)
        at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.metadata(EntityManagerFactoryBuilderImpl.java:847)
        at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.generateSchema(EntityManagerFactoryBuilderImpl.java:857)
        ... 26 more
Caused by: org.hibernate.engine.jndi.JndiException: Error parsing JNDI name [java:jboss/datasources/PostgreSqlDS]
        at org.hibernate.engine.jndi.internal.JndiServiceImpl.parseName(JndiServiceImpl.java:124)
        at org.hibernate.engine.jndi.internal.JndiServiceImpl.locate(JndiServiceImpl.java:95)
        at org.hibernate.engine.jdbc.connections.internal.DatasourceConnectionProviderImpl.configure(DatasourceConnectionProviderImpl.java:98)
        at org.hibernate.boot.registry.internal.StandardServiceRegistryImpl.configureService(StandardServiceRegistryImpl.java:94)
        at org.hibernate.service.internal.AbstractServiceRegistryImpl.initializeService(AbstractServiceRegistryImpl.java:217)
        at org.hibernate.service.internal.AbstractServiceRegistryImpl.getService(AbstractServiceRegistryImpl.java:189)
        at org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator.buildJdbcConnectionAccess(JdbcEnvironmentInitiator.java:145)
        at org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator.initiateService(JdbcEnvironmentInitiator.java:66)
        at org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator.initiateService(JdbcEnvironmentInitiator.java:35)
        at org.hibernate.boot.registry.internal.StandardServiceRegistryImpl.initiateService(StandardServiceRegistryImpl.java:88)
        at org.hibernate.service.internal.AbstractServiceRegistryImpl.createService(AbstractServiceRegistryImpl.java:234)
        ... 36 more
Caused by: javax.naming.NoInitialContextException: Need to specify class name in environment or system property, or as an applet parameter, or in an application resource file:  java.naming.factory.initial
        at org.hibernate.engine.jndi.internal.JndiServiceImpl.parseName(JndiServiceImpl.java:118)
        ... 46 more

```

```
--- configuration begin ---
[javax.persistence.schema-generation.database.action:none, javax.persistence.schema-generation.scripts.action:drop-and-create, javax.persistence.schema-generation.scripts.create-target:file:/path/build/classes/main/META-INF/sql/create.sql, javax.persistence.schema-generation.scripts.drop-target:file:/path/build/classes/main/META-INF/sql/drop.sql, javax.persistence.database-product-name:PostgreSQL, javax.persistence.database-major-version:9, javax.persistence.database-minor-version:5, javax.persistence.jdbc.driver:null, javax.persistence.jdbc.url:null, javax.persistence.jdbc.user:null, javax.persistence.jdbc.password:null, javax.persistence.schema-generation.create-source:metadata, javax.persistence.schema-generation.drop-source:metadata, eclipselink.persistencexml:META-INF/persistence.xml, eclipselink.weaving:false, hibernate.archive.autodetection:class,hbm, hibernate.dialect:org.hibernate.dialect.PostgreSQL94Dialect, datanucleus.persistenceXmlFilename:META-INF/persistence.xml, javax.persistence.schema-generation-connection:io.github.divinespear.gradle.plugin.ConnectionMock@52e408e1, javax.persistence.validation.mode:none, javax.persistence.transactionType:RESOURCE_LOCAL, javax.persistence.jtaDataSource:null, javax.persistence.nonJtaDataSource:null]
--- configuration end ---
```

**Comment by divinespear** [comment]
Sorry to late response, I'm on deadly hard working :/
I'll release this fix soon.

**Comment by divinespear** [comment]
released as 0.2.1 today. enjoy with it!
